ons-spark/spark-functions/logistic-regression.md

```diff
@@ -0,0 +1,1660 @@
+# Logistic Regression
+
+Logistic regression is often used in both predictive and inferential analysis. A logistic
+regression model is trained on a dataset to learn how certain
+independent variables relate to the probability of a binary outcome
+occurring. The model is then applied to cases where the independent
+variables are known but the outcome is unknown, and the probability of
+the outcome occurring can be estimated.
+
+This article shows how to use the `Spark ML` functions to generate a
+logistic regression model in PySpark and sparklyr, including a
+discussion of the steps necessary to prepare the data and potential
+issues to consider. We will also show how to access standard errors and
+confidence intervals for inferential analysis and how to assemble these steps into a pipeline.
+
+## Preparing the data
+
+We will be using the `Spark ML` functions `GeneralizedLinearRegression`/`ml_generalised_linear_regression`. The syntax used to call these
+functions is somewhat similar to logistic regression functions in python or R.
+However, there may be some additional steps to take in preparing the data
+before the model can be run successfully.
+
+We can read the data in as follows:
+
+````{tabs}
+```{code-tab} py 
+#import packages
+
+from pyspark.sql import SparkSession
+import pyspark.sql.functions as F
+
+import pandas as pd
+
+
+#set up small Spark session
+
+""" spark = (
+    SparkSession.builder.appName("pyspark_lab_2")
+    .config("spark.executor.memory", "1g")
+    .config("spark.executor.cores", 1)
+    .config("spark.dynamicAllocation.enabled", "true")
+    .config("spark.dynamicAllocation.maxExecutors", 3)
+    .config("spark.sql.shuffle.partitions", 12)
+    .config("spark.shuffle.service.enabled", "true")
+    .config("spark.ui.showConsoleProgress", "false")
+    .enableHiveSupport()
+    .getOrCreate()
+)
+ """
+
+#local session 
+spark = (SparkSession.builder.master("local[2]")
+         .appName("logistic-regression")
+
+# Set the data path
+rescue_path_parquet = '/training/rescue_clean.parquet'
```

Comment:
It would be consistent with the rest of the book to read in the config file and use that to get the paths to the datasets

Reply:
This is a good point. Unfortunately, the cleaned dataset I've used is a bit different to the regular rescue dataset that is in the config file. I could swap it for that dataset but there'd be some extra steps in the pre-processing so the data preparation part and possibly some other sections would have to be re-written quite a bit.

I'll speak to Wil about it but I'm wondering if it might be best for now to leave it how it is and raise the data swap or putting it in the config file as an issue we can fix later just so we can get the page up and running for now. What do you think?